### PR TITLE
CASMINST-4446: Fix goss-network-interfaces test to handle CHN and CAN configurations

### PR DESCRIPTION
**Pull request by johren-hpe**
## Summary and Scope

This test will always check to see if the bond0.can0 interface exists on the worker and master nodes.   However, this interface not exist if the system was configured with CHN instead of CAN.   This changes the test into a script that checks the BICAN network configuration in SLS to determine if it is CAN or CHN.   It will only check for the bond0.can0 interface if it is configured for CAN.

## Issues and Related PRs

* Resolves CASMINST-4446

### Tested on:

  * `surtur` and `wasp`

### Test description:

I ran this goss test on wasp where it is configured for CHN and previous failed.  It now passes there.
I also ran this goss test on surtur where it was configured for CAN to make sure it still passes there.

Wasp:
```
ncn-m001:~ # GOSS_BASE=/opt/cray/tests/install/ncn goss -g /opt/cray/tests/install/ncn/tests/goss-network-interfaces.yaml --vars=/opt/cray/tests/install/ncn/vars/variables-ncn.yaml validate
....

Total Duration: 0.283s
Count: 4, Failed: 0, Skipped: 0
```

I also ran through several injected negative test cases (not able to get secret, not able to pull token, etc.) to test the error checking.

## Pull Request Checklist

- [x] Version number(s) incremented, if applicable
- [x] Copyrights updated
- [x] License file intact
- [x] Target branch correct
- [ ] CHANGELOG.md updated
- [x] Testing is appropriate and complete, if applicable
- [ ] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

**Review by jacobsalmela — APPROVED** [review]

File: goss-testing/scripts/check_network_interface.sh [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+TOKEN=$(curl -s -k -S -d grant_type=client_credentials -d client_id=admin-client -d client_secret="$SECRET" https://api-gw-service-nmn.local/keycloak/realms/shasta/protocol/openid-connect/token | jq -r '.access_token') ||
+    err_exit 15 "Command pipeline failed with return code $?: curl -s -k -S -d grant_type=client_credentials -d client_id=admin-client -d client_secret=XXXXXX https://api-gw-service-nmn.local/keycloak/realms/shasta/protocol/openid-connect/token | jq -r '.access_token'"
+if [ -z "$TOKEN" ] || [ "$TOKEN" == null ]; then
+    err_exit 17 "Failed to retrieve token from https://api-gw-service-nmn.local/keycloak/realms/shasta/protocol/openid-connect/token"
+fi
+
+
+# check SLS Networks data for BiCAN toggle
+USER_NETWORK=$(curl -s -k -H "Authorization: Bearer ${TOKEN}" https://api-gw-service-nmn.local/apis/sls/v1/networks/BICAN | jq -r .ExtraProperties.SystemDefaultRoute | tr '[:upper:]' '[:lower:]')
+if [ -z "$USER_NETWORK" ] || [ "$USER_NETWORK" == null ]; then
```
[the remark this answers is not in the document]
Comment (johren-hpe) [reply goss-testing/scripts/check_network_interface.sh:59]: Will address in a later PR

File: goss-testing/scripts/check_network_interface.sh [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+{
+    local rc
+    rc=$1
+    shift
+    echo "ERROR: $*" 1>&2
+    echo "FAIL"
+    exit $rc
+}
+
+TESTIF=$1
```
[the remark this answers is not in the document]
Comment (johren-hpe) [reply goss-testing/scripts/check_network_interface.sh:42]: Will address in a later PR

File: goss-testing/scripts/check_network_interface.sh [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+# check SLS Networks data for BiCAN toggle
+USER_NETWORK=$(curl -s -k -H "Authorization: Bearer ${TOKEN}" https://api-gw-service-nmn.local/apis/sls/v1/networks/BICAN | jq -r .ExtraProperties.SystemDefaultRoute | tr '[:upper:]' '[:lower:]')
+if [ -z "$USER_NETWORK" ] || [ "$USER_NETWORK" == null ]; then
+    err_exit 18 "Failed to retrieve user network from SLS"
+fi
+
+if [[ "$TESTIF" =~ .*can0 ]]; then
+    if [[ "$TESTIF" =~ .*"$USER_NETWORK".* ]]; then
+        ip link show dev $TESTIF || err_exit 20 "Command pipeline failed with return code $?: ip link show dev $TESTIF"
+    fi
```
[the remark this answers is not in the document]
Comment (johren-hpe) [reply goss-testing/scripts/check_network_interface.sh:66]: Will address in a later PR

File: goss-testing/scripts/check_network_interface.sh [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    err_exit 12 "Failed to retrieve admin-client-auth secret"
+fi
+
+TOKEN=$(curl -s -k -S -d grant_type=client_credentials -d client_id=admin-client -d client_secret="$SECRET" https://api-gw-service-nmn.local/keycloak/realms/shasta/protocol/openid-connect/token | jq -r '.access_token') ||
+    err_exit 15 "Command pipeline failed with return code $?: curl -s -k -S -d grant_type=client_credentials -d client_id=admin-client -d client_secret=XXXXXX https://api-gw-service-nmn.local/keycloak/realms/shasta/protocol/openid-connect/token | jq -r '.access_token'"
+if [ -z "$TOKEN" ] || [ "$TOKEN" == null ]; then
+    err_exit 17 "Failed to retrieve token from https://api-gw-service-nmn.local/keycloak/realms/shasta/protocol/openid-connect/token"
+fi
+
+
```
[the remark this answers is not in the document]
Comment (johren-hpe) [reply goss-testing/scripts/check_network_interface.sh:56]: Will address in a later PR

**Pull request merged**